docs/migrations.md

```diff
@@ -0,0 +1,224 @@
+# Django and Wagtail Migrations
+
+Adding or changing fields on Django models, Wagtail page models
+(which are a particular kind of Django model), or StreamField block classes
+will always require a new [Django schema migration](#schema-migrations);
+additionally, changing field names or types on an existing block will require a
+[Django data migration](#data-migrations).
+
+
+## Table of contents
+
+1. [Reference material](#reference-material)
+1. [Schema migrations](#schema-migrations)
+1. [Data migrations](#data-migrations)
+   1. [Wagtail-specific consideration](#wagtail-specific considerations)
+   1. [Utility functions](#utility-functions)
+
+
+## Reference material
+
+The following links may be useful for setting context or diving deeper
+into the concepts presented throughout this page:
+
+- [Django migrations documentation](https://docs.djangoproject.com/en/1.11/topics/migrations/)
+- [Django data migrations documentation](https://docs.djangoproject.com/en/1.11/topics/migrations/#data-migrations)
+- [Wagtail Streamfield migrations documentation](https://docs.wagtail.io/en/v1.13.4/topics/streamfield.html#migrations)
+
+
+## Schema migrations
+
+Any time you add or change a field on a Django model, Wagtail page model
+(which are a particular kind of Django model), or StreamField block classes, a
+[Django schema migration](https://docs.djangoproject.com/en/1.11/topics/migrations)
+will be required.
+
+To automatically generate a schema migration,
+run the following, editing it to give your migration a name
+that briefly describes the change(s) you're making:
+
+```bash
+./cfgov/manage.py makemigrations -n <description_of_changes>
+```
+
+Be aware that each migration must have
+a unique four-digit number at the beginning of its filename,
+which is a regular source of conflicts between pull requests
+that are in flight at the same time.
+If a PR with a migration gets merged between the time you create your migration
+and the time that your PR is ready for merging,
+you will have to update your branch as normal to be current with master
```

Comment:
This paragraph is important and a good reference, I think it could be its own section, and some of the content could be better written as an ordered list with steps. It could come after the note below, with its own sub-heading here, e.g. "Git conflicts in migrations"

Reply:
This could also be done by a later PR

Reply:
Rearranged a bit as you suggested. I don't know how well it can be made into an ordered list, though. The steps would be pretty dependent on the particulars of a person's local setup.